core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/HistogramFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.bucket.histogram;
+
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.InternalAggregations;
+import org.elasticsearch.search.aggregations.bucket.MultiBucketsAggregation;
+
+import java.util.List;
+
+/** Implemented by histogram aggregations and used by pipeline aggregations to insert buckets. */
+public interface HistogramFactory {
+
+    Number getKey(MultiBucketsAggregation.Bucket bucket);
+
+    Number nextKey(Number key);
+
+    InternalAggregation createAggregation(List<MultiBucketsAggregation.Bucket> buckets);
+
+    MultiBucketsAggregation.Bucket createBucket(Number key, long docCount, InternalAggregations aggregations);
+
```

Comment:
can we doc the new methods here too, since they are on an interface, they represent a contract.